File: openci-runner/src/config/sentry.ts
import * as Sentry from "@sentry/cloudflare";

export const sentryConfig = (env: Env): Sentry.CloudflareOptions => ({
	dsn: env.SENTRY_DSN,
	enableLogs: true,
	environment: env.ENVIRONMENT,
	integrations: [
		Sentry.consoleLoggingIntegration({ levels: ["log", "warn", "error"] }),
	],
	tracesSampleRate: 1.0,
});

File: apps/build_job_executor/lib/src/sentry.dart
import 'package:sentry/sentry.dart';

Future<void> initializeSentry(String? sentryDsn) async {
  if (sentryDsn != null && sentryDsn.isNotEmpty) {
    await Sentry.init((options) {
      options.dsn = sentryDsn;
      options.tracesSampleRate = 1.0;
    });
  }
}
